src/slack/slack.js

```diff
@@ -255,22 +255,39 @@ export class Slack {
 
   async submitOrder(userId) {
     const order = this.orders[userId]
+    order.step = 'finished'
     const dbId = await this.storeOrder(order)
 
     await this.notifyOfficeManager(order, dbId, userId, order.isCompany)
-    await this.updateMessage(order, null, {
-      pretext: order.isCompany ? ':office: Company order finished:' : ':woman: Personal order finished:',
-      color: 'good',
-      actions: [],
-    })
+    // const msgTitle = order.isCompany ? ':office: Company order finished:' : ':woman: Personal order finished:'
+    // await this.updateMessage(order, msgTitle)
+    // try {
+    //   await this.boltApp.client.chat.postMessage({
+    //     channel: userId,
+    //     text: order.isCompany ? ':office: Company order finished :point_up:' : ':woman: Personal order finished :point_up:',
+    //   })
+    // } catch (err) {
+    //   logger.error(`Failed to post 'order finished' message to user '${userId}': ${err}`)
+    // }
+    const {channel, ts} = order.orderConfirmation
     try {
-      await this.boltApp.client.chat.postMessage({
+      await this.boltApp.client.chat.delete({channel, ts})
+    } catch (err) {
+      logger.error(`Failed to delete message on channel '${channel}': ${err}`)
+    }
+
+    const msgTitle = order.isCompany ? ':office: Company order finished:' : ':woman: Personal order finished:'
+    try {
+      const orderAttachment = this.userOrderAttachment(order, msgTitle)
```

Comment:
would it make sense to move this `msgTitle` decision inside the `userOrderAttachment`? probably not deep into `USER_STEP_MAP`, because we need both `step` and `isCompany` to decide the message... I wonder whether `USER_STEP_MAP` should actually be a function so we can base output on more than just `step` - we should have a `switch` clause inside such function and it would still be very readable

Reply:
can be done later of course

Reply:
sure it can be done, switch between `message` and `action`
We can do it in a separate task if you agree.